VirtualH89/Src/MemoryDecoder.h

```diff
@@ -0,0 +1,55 @@
+///
+/// \file MemoryDecoder.h
+///
+/// Base class for memory decoders, which implement various layouts of
+/// the address space along with switching between them.
+///
+/// \date Mar 05, 2016
+/// \author Douglas Miller
+///
+#ifndef MEMORYDECODER_H_
+#define MEMORYDECODER_H_
+
+#include "Memory8K.h"
+#include "MemoryLayout.h"
+#include "GppListener.h"
+
+class MemoryDecoder: public GppListener
+{
```

Comment:
The base MemoryDecoder should be generic, and not derive from GppListener. Then this one could be GppMemoryDecoder, which most of the current child classes could derive from. The H88MemoryDecoder would just derive from the new base class.